src/components/ui/coaching-sessions/editor-cache-context.tsx

```diff
@@ -113,27 +127,75 @@ export const EditorCacheProvider: React.FC<EditorCacheProviderProps> = ({
             color: "#ffcc00",
           });
 
-          setCache({
+          setCache(prev => ({
+            ...prev,
             yDoc: doc,
             collaborationProvider: provider,
             extensions: collaborativeExtensions,
             isReady: true,
             isLoading: false,
             error: null,
-          });
+          }));
         },
         onDisconnect: () => {
           console.log('🔌 Editor cache: Collaboration disconnected');
         },
       });
 
+      // Set user awareness with presence data using centralized role logic
+      const userPresence = createConnectedPresence({
+        userId: userSession.id,
+        name: userSession.display_name,
+        role: userRole,
+        color: "#ffcc00"
+      });
+
       provider.setAwarenessField("user", {
         name: userSession.display_name,
         color: "#ffcc00",
       });
+      
+      provider.setAwarenessField("presence", userPresence);
 
       providerRef.current = provider;
 
+      // Listen for awareness changes to track presence
+      provider.on('awarenessChange', ({ states }: { states: Map<string, any> }) => {
+        const updatedUsers = new Map<string, UserPresence>();
+        
+        states.forEach((state, clientId) => {
+          if (state.presence) {
+            const presence = toUserPresence(state.presence);
+            updatedUsers.set(presence.userId, presence);
+          }
+        });
```

Comment:
Error handling issue:

```typescript
states.forEach((state, clientId) => {
  if (state.presence) {
    const presence = toUserPresence(state.presence);
    updatedUsers.set(presence.userId, presence);
  }
});
```

- Missing error handling for malformed presence data
- toUserPresence could throw if awareness data is invalid

Reply:
Solved in ac11e5d85933cdf6e0bf2813ea05877eae58b81b  as well